### PR TITLE
[WIP]MB-926 introducing global error based flow control

### PR DESCRIPTION
**Pull request by ramith**
This is still a working progress.
this to do :
 1) Implement the isOperational for hector and CQL
 2) UI Integration
 3) Lots of testing.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java [hunk 1]
```diff
@@ -205,38 +209,67 @@ public static void shutDownAndesKernel() throws AndesException {
...
+     * 
+     * @return an implementation of {@link AndesContextStore}
+     * @throws Exception if an error occures
+     */
+    private static AndesContextStore createAndesContextStoreFromConfig() throws Exception {
+        StoreConfiguration andesConfiguration = AndesContext.getInstance()
+                .getStoreConfiguration();
+        //create a andes context store and register
+        String contextStoreClassName = andesConfiguration.getAndesContextStoreClassName();
+        Class<? extends AndesContextStore> contextStoreClass = Class.forName(contextStoreClassName).asSubclass(AndesContextStore.class);
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java]: can we break long statements into few lines?

Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java]: +1

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeliveryWorker.java [hunk 2]
```diff
@@ -51,8 +63,22 @@
...
+     * marked as volatile since this value could be set from a different thread
+     * (other than those of disrupter)
+     */
+    private volatile SettableFuture<Boolean> messageStoresUnavailable;
+
+    
     private static final long SLOT_DELETION_SCHEDULE_INTERVAL = 15 * 1000;
 
+    /**
+     * Maximum number to retries retrieve metadata list for a given storage queue ( in the errors occur in message stores)
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeliveryWorker.java]: format these docs properly

**Comment by ramith** [comment]
closing this one...
added few more stuff. in a different branch.

**Comment by ramith** [comment]
new one : https://github.com/wso2/andes/pull/117

**Pull request closed without merging**